pkg/apis/resolution/v1beta1/resolution_request_types.go

```diff
@@ -79,9 +80,44 @@ type ResolutionRequestStatusFields struct {
 	// of the requested resource in-lined into the ResolutionRequest
 	// object.
 	Data string `json:"data"`
+	// Source is the source reference of the remote data that records the url, digest
+	// and the entrypoint.
+	Source *ConfigSource `json:"source"`
 }
 
 // GetStatus implements KRShaped.
 func (rr *ResolutionRequest) GetStatus() *duckv1.Status {
 	return &rr.Status.Status
 }
+
+// ConfigSource are the source reference of the remote resource in the standard SLSA
+// format.
+// +k8s:deepcopy-gen=false
```

Comment:
This is because the autogenerated deepcopy functions will complain about that the vendor type wrapped in the `ConfigSource` doesn't have deepcopy functions. So we can write our own deepcopy functions by disabling the autogen.

Reply:
ok, do we really have to wrap the ConfigSource then? Its only like 3 fields anyway

Reply:
Right, there are only 3 fields in this struct. Defining our own configsource with the 3 fields is one of the options we explored before. I think the idea here was that intoto has this defined already and it is aligned with slsa spec, so thought it might be good to use the standard one, and also makes it a bit easier for Chains to grab the data? I don't have strong preference here.
Happy to change back to define our own configsource if we are leaning towards this approach.

cc @wlynch for thoughts.

Reply:
I'm fine with either approach.

If we fork the type make sure to document each field heavily - assume people don't know what SLSA / in-toto is and don't know what should go in each field (i.e. what's an entrypoint, why is digest a map, do values need a particular format, etc.)